### ISSUE TITLE
According to the paper, I have an confusion in cosine similarity function 

### ISSUE THREAD

**Issue by peterwisu** [Rudrabha/Wav2Lip#443]
So, the first question in the paper state," we compute a dot product between the ReLU-activated video and speech embeddings v,s to
yield a single value between [0, 1]"  as I understand the use of ReLU, the last layer does limit the range of the similarity score from cosine similarity to be in a range from 0 to 1 instead of -1 to 1, right? since all the negative value in the embedding will be 0 make the angle maximum at 90 degrees right?.

The second question is in the code in wav2lip/models/syncnet.py. After both embeddings is obtained, the Euclidean norm is applied to both embeddings and returns both values as given in this line.

`audio_embedding = F.normalize(audio_embedding, p=2, dim=1) `
`face_embedding = F.normalize(face_embedding, p=2, dim=1)`
`return audio_embedding, face_embedding`

After that, the cosine similarity uses these two embeddings to calculate the similarity score. But the equation in the paper shows differently,

![Screenshot from 2023-01-30 20-11-21](https://user-images.githubusercontent.com/94657890/215584687-dc59596e-b847-4304-85a4-48c10d520d63.png)
why does only the denominator in the equation have the Euclidean norm notation, but the above dot product does not? Did I understand something wrong? Thank you
